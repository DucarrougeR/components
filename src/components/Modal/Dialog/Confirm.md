Wrap Confirm around a component whose action needs confirmation, e.g. a delete button.

```js
import { Confirm } from './Confirm'
import { Button } from '../../Button'

const handleConfirm = close => {
  alert('You did something')
  close()
}
;<Confirm
  title="Confirm Something"
  message="Is this what you want to do?"
  onConfirm={handleConfirm}
>
  {open => <Button onClick={open}>Do Something</Button>}
</Confirm>
```
