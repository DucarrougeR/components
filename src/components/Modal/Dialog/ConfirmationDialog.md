If your usage does not lend itself to the render prop style (e.g. the confirmation is triggered from inside a hook), use `<ConfirmationDialog/>` and control it directly with the `isOpen` and `close` props.

```js
import { useToggle } from '../../../hooks'
import { ConfirmationDialog } from './ConfirmationDialog'
import { Button } from '../../Button'
import { Paragraph } from '../../Text/Paragraph'

const Component = () => {
  const [keyPressed, setKeypressed] = React.useState('none')
  const { value, setOn, setOff } = useToggle(false)

  function handleConfirm() {
    alert('You deleted it')
    setOff()
  }

  React.useEffect(() => {
    function onKeydown() {
      const key = event.keyCode || event.charCode
      if (key === 8 || key === 46) {
        setKeypressed(key === 8 ? 'Backspace' : 'Delete')
        setOn()
      }
    }
    window.addEventListener('keydown', onKeydown)
    return () => {
      window.removeEventListener('keydown', onKeydown)
    }
  }, [])

  return (
    <ConfirmationDialog
      isOpen={value}
      close={setOff}
      confirmLabel={'Yes, Delete'}
      color="danger"
      title={`You pressed ${keyPressed}`}
      message="Are you sure you want to delete it?"
      onConfirm={handleConfirm}
    />
  )
}
;<>
  <Component />
  <Paragraph>Hit the delete or backspace key.</Paragraph>
</>
```
