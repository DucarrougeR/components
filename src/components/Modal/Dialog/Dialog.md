Dialogs inform users about a task and can contain critical information, require decisions, or involve multiple tasks.

```js
<Dialog content="This is some content in the Dialog">
  <Button>Open Dialog</Button>
</Dialog>
```

---

### Simple Confirmation Dialog

A confirmation dialog can be easily assembled using the Dialog and its sibling components DialogHeader and DialogFooter.

```js
const ModalContext = require('../ModalContext').ModalContext

const content = (
  <>
    <ModalHeader>
      <Heading>Are you sure you want to delete "Stuff"?</Heading>
    </ModalHeader>
    <ModalContext.Consumer>
      {({ closeModal }) => (
        <ModalFooter>
          <Button onClick={closeModal} variant="transparent" mr="small">
            Cancel
          </Button>
          <Button
            onClick={() => {
              alert("doin' things...")
              closeModal && closeModal()
            }}
            color="danger"
          >
            Yes, Delete "Stuff"
          </Button>
        </ModalFooter>
      )}
    </ModalContext.Consumer>
  </>
)
;<Dialog content={content}>
  <Button variant="outline" color="danger">
    Delete Stuff
  </Button>
</Dialog>
```
