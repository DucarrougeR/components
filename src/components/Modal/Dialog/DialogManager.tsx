import * as React from 'react'
import { Dialog } from '.'
import { ManagedModalProps } from '../Modal'
import { ModalManager, ModalManagerProps } from '../ModalManager'

export class DialogManager extends ModalManager<ModalManagerProps> {
  protected renderModal(content: React.ReactNode, props: ManagedModalProps) {
    return (
      <Dialog isOpen={this.state.isOpen} onClose={this.close} {...props}>
        {content}
      </Dialog>
    )
  }
}
