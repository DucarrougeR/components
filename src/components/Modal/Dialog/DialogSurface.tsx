import * as React from 'react'
import { styled } from '../../../style'
import { CustomizableModalAttributes } from '../Modal'
import { ModalSurface, ModalSurfaceProps } from '../ModalSurface'

export const DialogSurface: React.FC<ModalSurfaceProps> = ({
  children,
  ...props
}) => {
  return (
    <Surface
      borderRadius={CustomizableModalAttributes.surface.borderRadius}
      maxHeight="90vh"
      maxWidth="100%"
      {...props}
    >
      {children}
    </Surface>
  )
}

const Surface = styled(ModalSurface)`
  &.entering,
  &.exiting {
    opacity: 0.01;
    transform: translateY(100%);
  }

  &.exited {
    opacity: 1;
    transform: translateY(0%);
  }
`
