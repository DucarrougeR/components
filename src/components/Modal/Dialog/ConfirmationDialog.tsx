import * as React from 'react'
import { SemanticColor, SemanticColors } from '../../../style'
import { Button } from '../../Button'
import { Heading } from '../../Heading'
import { Paragraph } from '../../Text'
import { ModalContent, ModalFooter, ModalHeader } from '../Layout'
import { ManagedModalProps } from '../Modal'
import { Dialog } from './Dialog'

export type ConfirmationCallback = (close: () => void) => void

export interface ConfirmationProps extends ManagedModalProps {
  /**
   * Cancel button text
   * @default 'Cancel'
   */
  cancelLabel?: string
  /**
   * Defines the color of the confirm button. Can be the string name of a color listed in the color theme, or a color object.
   * @default "primary"
   */
  buttonColor?: keyof SemanticColors | SemanticColor
  /**
   * Confirmation button text
   * @default 'Confirm'
   */
  confirmLabel?: string
  /**
   * Additional information about the action requiring confirmation
   */
  message: string
  /**
   * Callback if user clicks Cancel button or closes the dialog
   */
  onCancel?: ConfirmationCallback
  /**
   * Function called when user clicks to confirm
   * close function is passed as an argument to control when to close the dialog
   */
  onConfirm: ConfirmationCallback
  /**
   * Dialog title text
   */
  title: string
}

export interface ConfirmationDialogProps extends ConfirmationProps {
  /**
   * For triggering close from within the dialog
   */
  close: () => void
  /**
   * Toggling this after mounting will trigger the animation
   * @default false
   */
  isOpen?: boolean
}

export const ConfirmationDialog: React.FC<ConfirmationDialogProps> = ({
  cancelLabel = 'Cancel',
  close,
  confirmLabel = 'Confirm',
  buttonColor = 'primary',
  isOpen = false,
  message,
  onCancel,
  onConfirm,
  title,
  ...props
}) => {
  const confirm = React.useCallback(() => {
    onConfirm(close)
  }, [onConfirm])

  const cancel = React.useCallback(() => {
    if (onCancel) {
      onCancel(close)
    } else {
      close()
    }
  }, [onCancel])

  return (
    <Dialog isOpen={isOpen} onClose={cancel} {...props}>
      <ModalHeader>
        <Heading>{title}</Heading>
      </ModalHeader>
      {message && (
        <ModalContent>
          <Paragraph>{message}</Paragraph>
        </ModalContent>
      )}
      <ModalFooter>
        <Button onClick={confirm} color={buttonColor}>
          {confirmLabel}
        </Button>
        <Button onClick={cancel} variant="transparent">
          {cancelLabel}
        </Button>
      </ModalFooter>
    </Dialog>
  )
}
