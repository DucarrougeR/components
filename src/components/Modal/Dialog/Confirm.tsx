import * as React from 'react'
import { useToggle } from '../../../hooks'
import { ConfirmationDialog, ConfirmationProps } from './ConfirmationDialog'

export interface ConfirmProps extends ConfirmationProps {
  /**
   * Render prop is passed the confirmation opener
   */
  children: (open: () => void) => React.ReactNode
}

export const Confirm: React.FC<ConfirmProps> = ({ children, ...props }) => {
  const { value, setOn, setOff } = useToggle()

  return (
    <>
      {children(setOn)}
      <ConfirmationDialog isOpen={value} close={setOff} {...props} />
    </>
  )
}
