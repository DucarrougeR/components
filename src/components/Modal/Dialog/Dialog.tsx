import * as React from 'react'
import { Modal, ModalProps } from '../Modal'
import { DialogSurface } from './DialogSurface'

export interface DialogProps extends ModalProps {
  /**
   * Specifying a width will force the DrawerSurface to be the lesser of the specified width or the viewport width.
   * You can also specify `auto` if you want the Drawer to auto-size to its contents.
   * @default auto
   */
  width?: string
  /**
   * Content that will be placed inside the Dialog
   * @required
   */
  content: React.ReactNode
}

export const Dialog: React.SFC<DialogProps> = ({
  width,
  content,
  children,
  surfaceStyles,
  ...props
}) => {
  const surface = (className: string) => {
    return (
      <DialogSurface style={surfaceStyles} className={className} width={width}>
        {content}
      </DialogSurface>
    )
  }

  return (
    <Modal render={surface} {...props}>
      {children}
    </Modal>
  )
}
