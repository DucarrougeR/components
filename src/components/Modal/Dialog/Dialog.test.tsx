import { ReactWrapper } from 'enzyme'
import 'jest-styled-components'
import * as React from 'react'
import { assertSnapshot } from '../../../../test/utils/snapshot'

import { mountWithTheme } from '../../../../test/utils/create_with_theme'

import { Button } from '../../Button'
import { Heading } from '../../Heading'
import { ModalFooter, ModalHeader } from '../Layout'
import { SimpleContent } from '../modal.test.helpers'

import { ModalBackdrop } from '../ModalBackdrop'
import { ModalContainer } from '../ModalContainer'
import { ModalSurface } from '../ModalSurface'
import { Dialog } from './Dialog'

test('Dialog Hidden', () => {
  assertSnapshot(
    <Dialog content="words and stuff">
      <Button>ðŸ¥‘</Button>
    </Dialog>
  )
})

test('Dialog Shown', () => {
  assertSnapshot(<Dialog content={SimpleContent} open />)
})

test('Dialog, backdrop customized', () => {
  assertSnapshot(
    <Dialog
      open
      content={SimpleContent}
      backdropStyles={{ background: 'purple', opacity: 1 }}
    >
      <Button>ðŸ¥‘</Button>
    </Dialog>
  )
})

test('Dialog opens on click', () => {
  const dialog = mountWithTheme(
    <Dialog content={SimpleContent}>
      <Button>Open Modal</Button>
    </Dialog>
  )

  expect(dialog.find(ModalContainer).exists()).toEqual(false)
  dialog.find('button').simulate('click')
  expect(dialog.find(ModalContainer).exists()).toEqual(true)

  const backdrop = dialog.find(ModalBackdrop)
  expect(backdrop.exists()).toEqual(true)
  backdrop.simulate('click')

  expect(dialog.find(ModalContainer).exists()).toEqual(false)
})

test('contains the content passed to it', () => {
  const dialog = mountWithTheme(
    <Dialog content={SimpleContent}>
      <Button>Open Modal</Button>
    </Dialog>
  )

  dialog.find('button').simulate('click') // Click to open
  expect(dialog.contains(SimpleContent)).toBeTruthy()
})

describe('Dialog Styling', () => {
  let dialog: ReactWrapper
  beforeEach(() =>
    (dialog = mountWithTheme(
      <Dialog
        open
        content={SimpleContent}
        backdropStyles={{ backgroundColor: 'pink' }}
        surfaceStyles={{ backgroundColor: 'purple' }}
      >
        <Button>Open Modal</Button>
      </Dialog>
    )))
  afterEach(() => dialog.unmount())

  test('Dialog applies the backdrop styles', () => {
    const backdrop = dialog.find(ModalBackdrop)
    expect(backdrop.props().style).toEqual({ backgroundColor: 'pink' })
  })

  test('Dialog applies the surface styles', () => {
    const surface = dialog.find(ModalSurface)
    expect(surface.props().style).toEqual({ backgroundColor: 'purple' })
  })
})

test('Confirmation Dialog, Shown', () => {
  assertSnapshot(
    <Dialog
      open
      content={
        <>
          <ModalHeader>
            <Heading>Are you sure you want to delete "Stuff"?</Heading>
          </ModalHeader>
          <ModalFooter>
            <Button variant="transparent" mr="medium">
              Cancel
            </Button>
            <Button color="danger">Yes, Delete "Stuff"</Button>
          </ModalFooter>
        </>
      }
    >
      <Button variant="outline" color="danger">
        Delete Stuff
      </Button>
    </Dialog>
  )
})
