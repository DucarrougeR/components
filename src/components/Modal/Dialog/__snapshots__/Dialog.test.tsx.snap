// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Dialog, Shown 1`] = `
Array [
  .c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: #FFFFFF;
  border-color: #C1C6CC;
  color: #CC1F36;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(204,31,54,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #FFFFFF;
  border-color: #C1C6CC;
  color: #CC1F36;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #FFFFFF;
  border-color: #CC1F36;
  color: #990F14;
}

.c0:active,
.c0.active {
  background: #CC1F36;
  border-color: #CC1F36;
  color: #FFFFFF;
}

<button
    className="c0 c1"
    onClick={[Function]}
    style={
      Object {
        "lineHeight": "34px",
      }
    }
  >
    Delete Stuff
  </button>,
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  background-color: #DEE1E5;
  height: 100%;
  opacity: 0.6;
  position: fixed;
  width: 100%;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border-radius: 0.25rem;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 90vh;
  max-width: 100%;
  position: absolute;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 1.25rem;
}

.c7 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-right: 2rem;
}

.c9 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.75rem;
}

.c11 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1.375rem;
  margin-left: auto;
  padding: 0rem;
}

.c12 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1em;
  width: 1em;
}

.c13 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 1.25rem;
}

.c14 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  margin-right: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c16 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c10 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: transparent;
  border-color: transparent;
  color: #64518A;
}

.c10:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c10[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c10[disabled]:hover,
.c10[disabled]:active,
.c10[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

.c10:hover,
.c10:focus,
.c10.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c10:active,
.c10.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

.c15 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: #CC1F36;
  border-color: #CC1F36;
  color: #FFFFFF;
}

.c15:focus {
  box-shadow: 0 0 0 0.15rem rgba(204,31,54,0.25);
}

.c15[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c15[disabled]:hover,
.c15[disabled]:active,
.c15[disabled]:focus {
  background-color: #CC1F36;
  border-color: #C1C6CC;
}

.c15:hover,
.c15:focus,
.c15.hover {
  background: #B2121F;
  border-color: #B2121F;
}

.c15:active,
.c15.active {
  background: #990F14;
  border-color: #990F14;
}

.c8 {
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-transform: none;
}

.c1 {
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  cursor: pointer;
}

.c1.entering,
.c1.exiting {
  opacity: 0.01;
}

.c1.exited {
  opacity: 0.6;
}

.c4 {
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  -webkit-transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
}

.c3.entering,
.c3.exiting {
  opacity: 0.01;
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.c3.exited {
  opacity: 1;
  -webkit-transform: translateY(0%);
  -ms-transform: translateY(0%);
  transform: translateY(0%);
}

<div
    className="c0"
  >
    <div
      className="faux c1 c2"
      onClick={[Function]}
    />
    <div
      className="faux c3 c4 c5"
    >
      <header
        className="c6"
      >
        <div
          className="c7"
        >
          <h2
            className="c8 c9"
          >
            Are you sure you want to delete "Stuff"?
          </h2>
        </div>
        <button
          className="c10 c11"
          onClick={[Function]}
          style={
            Object {
              "lineHeight": 1,
            }
          }
        >
          <div
            className="c12"
          >
            <svg
              fill="currentColor"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
                fill="currentColor"
              />
            </svg>
          </div>
        </button>
      </header>
      <footer
        className="c13"
      >
        <button
          className="c10 c14"
          style={
            Object {
              "lineHeight": "34px",
            }
          }
        >
          Cancel
        </button>
        <button
          className="c15 c16"
          style={
            Object {
              "lineHeight": "34px",
            }
          }
        >
          Yes, Delete "Stuff"
        </button>
      </footer>
    </div>
  </div>,
]
`;

exports[`Dialog Hidden 1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<button
  className="c0 c1"
  onClick={[Function]}
  style={
    Object {
      "lineHeight": "34px",
    }
  }
>
  🥑
</button>
`;

exports[`Dialog Shown 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  background-color: #DEE1E5;
  height: 100%;
  opacity: 0.6;
  position: fixed;
  width: 100%;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border-radius: 0.25rem;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 90vh;
  max-width: 100%;
  position: absolute;
}

.c1 {
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  cursor: pointer;
}

.c1.entering,
.c1.exiting {
  opacity: 0.01;
}

.c1.exited {
  opacity: 0.6;
}

.c4 {
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  -webkit-transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
}

.c3.entering,
.c3.exiting {
  opacity: 0.01;
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.c3.exited {
  opacity: 1;
  -webkit-transform: translateY(0%);
  -ms-transform: translateY(0%);
  transform: translateY(0%);
}

<div
  className="c0"
>
  <div
    className="faux c1 c2"
    onClick={[Function]}
  />
  <div
    className="faux c3 c4 c5"
  >
    <div>
      simple content
    </div>
  </div>
</div>
`;

exports[`Dialog, backdrop customized 1`] = `
Array [
  .c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<button
    className="c0 c1"
    onClick={[Function]}
    style={
      Object {
        "lineHeight": "34px",
      }
    }
  >
    🥑
  </button>,
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  background-color: #DEE1E5;
  height: 100%;
  opacity: 0.6;
  position: fixed;
  width: 100%;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border-radius: 0.25rem;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 90vh;
  max-width: 100%;
  position: absolute;
}

.c1 {
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  cursor: pointer;
  background: purple;
  opacity: 1;
}

.c1.entering,
.c1.exiting {
  opacity: 0.01;
}

.c1.exited {
  opacity: 0.6;
}

.c4 {
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  -webkit-transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
}

.c3.entering,
.c3.exiting {
  opacity: 0.01;
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.c3.exited {
  opacity: 1;
  -webkit-transform: translateY(0%);
  -ms-transform: translateY(0%);
  transform: translateY(0%);
}

<div
    className="c0"
  >
    <div
      className="faux c1 c2"
      onClick={[Function]}
    />
    <div
      className="faux c3 c4 c5"
    >
      <div>
        simple content
      </div>
    </div>
  </div>,
]
`;
